src/backend/aspen/database/models/sample.py

```diff
@@ -90,7 +90,16 @@ class Sample(idbase, DictMixin):  # type: ignore
     """A physical sample.  Multiple sequences can be taken of each physical sample."""
 
     __tablename__ = "samples"
-    __table_args__ = (UniqueConstraint("submitting_group_id", "private_identifier"),)
+    __table_args__ = (
+        # Unique index uses default name format of `uq_samples_submitting_group_id`
```

Comment:
Vaguely interesting SQLAlchemy thing here:
The way that SQLAlchemy auto names (`naming_convention` boilerplate) unique indices only uses the table name and first column called out in unique index, just ignores later columns when creating the name. But because we already have a group-private unique index, that took the same name that the group-public unique index would use. So I had to manually give the new one a `name` for the alembic auto-generation to work (and for SQlAlchemy to be able to understand the uniqueness mapping, but I don't know if that really gets used or not...)